### PR TITLE
feat(Core/AI): CU_SAI - Option to set creature in combat with zone outside of a dungeon

### PR DESCRIPTION
**Pull request by P-Kito**
## CHANGES PROPOSED:
-  It is now possible to use the action SMART_ACTION_SET_IN_COMBAT_WITH_ZONE outside of a dungeon by with using a range

## TESTS PERFORMED:
- Was on a 2k player live server.

## HOW TO TEST THE CHANGES:

- Create a simple SAI with action 38 and enter a range as first param
- Spawn it outside a dungeon (open world)
- Check if it goes IC with you and not the whole map ;-)

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScriptMgr.h [hunk 1]
```diff
@@ -457,7 +457,7 @@ enum SMART_ACTION
     SMART_ACTION_EVADE                              = 24,     // No Params
     SMART_ACTION_FLEE_FOR_ASSIST                    = 25,     // With Emote
     SMART_ACTION_CALL_GROUPEVENTHAPPENS             = 26,     // QuestID
-    SMART_ACTION_COMBAT_STOP                        = 27,     // 
+    SMART_ACTION_COMBAT_STOP                        = 27,     //
```
Comment (FrancescoBorzi) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.h:460]: should mention about the new param

Comment (P-Kito) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h:460]: I didnt do that, my VSC just remove the blank space

Comment (FrancescoBorzi) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h:460]: yes, that is fine. What I meant is that we should also add a comment about the new parameter

> enter a range as first param

Comment (FrancescoBorzi) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h]: @P-Kito I meant in `SMART_ACTION_SET_IN_COMBAT_WITH_ZONE`, that's where you added a new param, right?

Comment (P-Kito) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h]: resolved :p

**Comment by P-Kito** [comment]
[x] tested - can be merged

**Comment by FrancescoBorzi** [comment]
@P-Kito conflict :P

**Comment by P-Kito** [comment]
I thought u would resolve it when merging? Its just the stupid comment xD

**Comment by boom8866** [comment]
Collapse exists SmartScript::ProcessAction

**Comment by FrancescoBorzi** [comment]
@boom8866  what?

**Comment by P-Kito** [comment]
> 
> 
> Collapse exists SmartScript::ProcessAction

You mean Crash? Share logs? Contact me on Discord: Kito    on Azerothcore Discord

**Pull request merged**